src/js/components/FrameworkConfiguration.js

```diff
@@ -0,0 +1,381 @@
+      formErrors,
+      isInitialDeploy,
+      formData
+    } = this.props;
+
+    if (reviewActive) {
+      const fileName = "config.json";
+      const configString = JSON.stringify(formData, null, 2);
+      const ieDownloadConfig = () => {
+        // Download if on IE
+        if (global.navigator.msSaveOrOpenBlob) {
+          const blob = new Blob([configString], { type: "application/json" });
+          global.navigator.msSaveOrOpenBlob(blob, fileName);
+        }
+      };
+
+      const renderKeys = {};
+      this.getHashMapRenderKeys(formData, renderKeys);
+
+      // todo, truncate if over three lines
+      const preInstallNotes = packageDetails.getPreInstallNotes();
+      let preinstall = null;
+      if (preInstallNotes && isInitialDeploy) {
+        const preInstallNotesParsed = StringUtil.parseMarkdown(preInstallNotes);
+        preInstallNotesParsed.__html =
+          "<strong>Preinstall Notes: </strong>" + preInstallNotesParsed.__html;
+
+        preinstall = (
+          <div
+            dangerouslySetInnerHTML={preInstallNotesParsed}
+            className="flush-bottom message message-warning"
+          />
+        );
+      }
+
+      let errorsAlert = null;
+      if (deployErrors) {
+        errorsAlert = <CosmosErrorMessage error={deployErrors} />;
+      }
+
+      return (
+        <div className="flex-item-grow-1">
+          <div className="container container-wide">
+            {errorsAlert}
+            {preinstall}
+            <div className="row">
+              <div className="column-4">
+                <h1 className="flush-top">Configuration</h1>
+              </div>
+              <div className="column-8 text-align-right">
+                <button
+                  className="button button-primary-link button-inline-flex"
+                  onClick={this.onEditConfigurationButtonClick.bind(this)}
+                >
+                  <Icon id="pencil" size="mini" family="system" />
+                  <span className="form-group-heading-content">
+                    {"Edit Config"}
+                  </span>
+                </button>
+                <a
+                  className="button button-primary-link flush-right"
+                  download={fileName}
+                  onClick={ieDownloadConfig}
+                  href={`data:attachment/json;content-disposition=attachment;filename=${fileName};charset=utf-8,${encodeURIComponent(configString)}`}
+                >
+                  <Icon id="download" size="mini" family="system" />
+                  <span className="form-group-heading-content">
+                    {"Download Config"}
+                  </span>
+                </a>
+              </div>
+            </div>
+            <HashMapDisplay
+              hash={formData}
+              renderKeys={renderKeys}
+              onEditClick={this.onReviewConfigurationRowClick.bind(this)}
+              headlineClassName={"text-capitalize"}
+              emptyValue={"\u2014"}
```

Comment:
https://github.com/dcos/dcos-ui/pull/2447/files#diff-c7a540e5f0f308737396e460f6eb98cdR1

Reply:
Nice, I will use that component once your PR merges. For now, I'll add this as a todo in this code.